engine/src/main/java/io/camunda/zeebe/engine/state/immutable/DistributionState.java

```diff
@@ -7,4 +7,24 @@
  */
 package io.camunda.zeebe.engine.state.immutable;
 
-public interface DistributionState {}
+import io.camunda.zeebe.protocol.impl.record.value.distribution.CommandDistributionRecord;
+
+public interface DistributionState {
+
+  /**
+   * Returns whether there are any distributions pending for a given key.
+   *
+   * @param distributionKey the key of the distribution
+   * @return true if there are pending distributions for the given key, otherwise false
+   */
+  boolean hasPendingDistribution(long distributionKey);
+
+  /**
+   * Returns the {@link CommandDistributionRecord} for the given distribution key
+   *
+   * @param distributionKey the key of the distribution
+   * @param partition the partition to distribute to
+   * @return an new instance of the {@link CommandDistributionRecord}
+   */
+  CommandDistributionRecord getCommandDistributionRecord(long distributionKey, int partition);
```

Comment:
💭 Can be left until later when the Redistributor is re-implemented, but we will likely also need a method to visit pending distributions for this.

Reply:
Fair enough, it can't hurt having it already though 🤷